src/commands/close.ts

```diff
@@ -1,84 +1,126 @@
-//
-// ________________________________________________________________________________________________
+/*
+This file is part of Needle.
+
+Needle is free software: you can redistribute it and/or modify it under the terms of the GNU
+Affero General Public License as published by the Free Software Foundation, either version 3 of
+the License, or (at your option) any later version.
+
+Needle is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along with Needle.
+If not, see <https://www.gnu.org/licenses/>.
+*/
 
 import { SlashCommandBuilder } from "@discordjs/builders";
-import { type CommandInteraction, GuildMember, type MessageComponentInteraction, Permissions, type ThreadChannel } from "discord.js";
+import {
+	CommandInteraction,
+	GuildMember,
+	MessageComponentInteraction,
+	Permissions,
+	ThreadChannel,
+} from "discord.js";
 import { shouldArchiveImmediately } from "../helpers/configHelpers";
-import { interactionReply, getMessage, getThreadAuthor } from "../helpers/messageHelpers";
+import {
+	interactionReply,
+	getMessage,
+	getThreadAuthor,
+} from "../helpers/messageHelpers";
 import { setEmojiForNewThread } from "../helpers/threadHelpers";
 import type { NeedleCommand } from "../types/needleCommand";
 
 export const command: NeedleCommand = {
 	name: "close",
-	shortHelpDescription: "Closes a thread by setting the auto-archive duration to 1 hour",
-	longHelpDescription: "The close command sets the auto-archive duration to 1 hour in a thread.\n\nWhen using auto-archive, the thread will automatically be archived when there have been no new messages in the thread for one hour. This can be undone by a server moderator by manually changing the auto-archive duration back to what it was previously, using Discord's own interface.",
+	shortHelpDescription:
+		"Closes a thread by setting the auto-archive duration to 1 hour",
+	longHelpDescription:
+		"The close command sets the auto-archive duration to 1 hour in a thread.\n\nWhen using auto-archive, the thread will automatically be archived when there have been no new messages in the thread for one hour. This can be undone by a server moderator by manually changing the auto-archive duration back to what it was previously, using Discord's own interface.",
 
 	async getSlashCommandBuilder() {
 		return new SlashCommandBuilder()
 			.setName("close")
-			.setDescription("Closes a thread by setting the auto-archive duration to 1 hour")
+			.setDescription(
+				"Closes a thread by setting the auto-archive duration to 1 hour"
+			)
 			.toJSON();
 	},
 
-	async execute(interaction: CommandInteraction | MessageComponentInteraction): Promise<void> {
+	async execute(
+		interaction: CommandInteraction | MessageComponentInteraction
+	): Promise<void> {
```

Comment:
This one is yikes to me as well

Reply:
See https://github.com/MarcusOtter/discord-needle/pull/102#discussion_r838882420